### ISSUE TITLE
Proposal: user domain whitelist

### ISSUE THREAD

**Issue by northys** [safing/portmaster#275]
**What would you like to be added**:

Some sort of textarea in config where I add domains I have to use instead of randomly disabling random blocklists provided by you without even knowing what's inside.

```
safing.io - unblocks safing.io only
*.aws.cloudflont.com - unblocks ANYTHING.aws.cloudfront.com
```

I wouldn't add more logic and complexity to this feature. 

**Nice to have:**

This user whitelist should allow anything no matter what. E.g. it should allow response of IP from private subnet for nuc.vpn.northys.cz without allowing this (there is something called split tree whatever that needs to be disabled).

**:fire: Killer feature:**

Every domain that was blocked for n times (definitely not only once I would love to configure this before we find the right magic number that fits everybody) should trigger a prompt `Do you wan't to whitelist fonts.google.com for /usr/bin/firefox?`. I mean prompt in the dashboard's sidebar not system notification (which won't work for fedora anyway) so when I open portmaster because "the website looks broken" I can see immediately that it is because portmaster blocked fonts.google.com or bootstrapcdn.com or something similar which is hotlinked a lot.

**Why is this needed**:

When security tools become pain in the ass it forces users to disable it's features which leads to weaker security. This is exactly what happened past month when I investigated why t.co, fonts.google.com and some other domains are blocked. I just disabled random blocklist to continue using my PC without even caring what is it supposed to block.

**Comment by dhaavi** [comment]
> config where I add domains I have to use instead of randomly disabling random blocklists

I think that the "Outgoing Rules" setting is pretty much what you want. It works as suggested, but let me tell you an interesting thing: If you enter `.aws.cloudflont.com` it will match both `aws.cloudflont.com` and `ANYTHING.aws.cloudflont.com`.

This is also the place where prompt decisions are persisted to.

> without even knowing what's inside. [the blocklist?]

Don't you find the block lists descriptions helpful? What would you expect to find out about the lists?

> This user whitelist should allow anything no matter what. E.g. it should allow response of IP from private subnet for nuc.vpn.northys.cz without allowing this (there is something called split tree whatever that needs to be disabled).

We are currently implementing a change that might result in this behavior. To early to tell. But I will keep this in mind.

> Every domain that was blocked for n times should trigger a prompt

This is a very interesting idea. Kind of like an adaptive prompting, where the Portmaster tries to detect if the user is trying to access a blocked domain. Could be a bit tricky to implement, but I'll have a chat with the team about that.
Internal note: tracked by CC#1869.

**Comment by northys** [comment]
> I think that the "Outgoing Rules" 

I'll try it! Thanks!

> Don't you find the block lists descriptions helpful? What would you expect to find out about the lists?

I didn't mean it against portmaster. There is no description to describe list of 50k blocked domains :) And there is currently only one option - unblock / block all of them.

> Internal note: tracked by CC#1869.

Just curious - what tool do you use to track tickets internally?

**Comment by davegson** [comment]
> I didn't mean it against portmaster. There is no description to describe list of 50k blocked domains :) And there is currently only one option - unblock / block all of them.

I agree it should not be that binary. Let us know if the "Outgoing Rules" worked - that is the place to whitelist some over-blocking of the lists as you describes.

For further reading you can have a [look at the decision process](https://docs.safing.io/portmaster/architecture/core-service/privacy-filter/#connection-evaluation-stages) of the Portmaster. It evaluates every connection from top to bottom and the first answer it gets is followed without further continuing down the list. So iE there you can see that the whitelisting of Outgoing Rules is stronger than the blocking of the Filter Lists.

> Just curious - what tool do you use to track tickets internally?

We use [OpenProject](https://www.openproject.org/) for our project management
